llvm/include/llvm/Analysis/GenericDomTreeUpdater.h

```diff
@@ -243,6 +267,35 @@ class GenericDomTreeUpdater {
   /// Drop all updates applied by all available trees and delete BasicBlocks if
   /// all available trees are up-to-date.
   void dropOutOfDateUpdates();
+
+private:
+  /// Helper structure used to hold all the basic blocks
+  /// involved in the split of a critical edge.
+  struct CriticalEdge {
+    BasicBlockT *FromBB;
+    BasicBlockT *ToBB;
+    BasicBlockT *NewBB;
+  };
+
+  /// Pile up all the critical edges to be split.
+  /// The splitting of a critical edge is local and thus, it is possible
+  /// to apply several of those changes at the same time.
+  SmallVector<CriticalEdge, 32> CriticalEdgesToSplit;
```

Comment:
How did you pick the small size of 32?

Reply:
This is copied from abea99f65a97248974c02a5544eaf25fc4240056.

Reply:
Would you be able to compile a few representative programs (e.g., clang) and print the maximum and average size? Otherwise I'd stick to the default small size.

Reply:
Tested with bzip2:
<details>
```sh
maximum: 92
average: 8.70909
```
Most will not exceed 7.
</details>

Reply:
Then I'd stick to the default value